### ISSUE TITLE
Add support for finagle-oauth2 library

### ISSUE THREAD

**Issue by vkostyukov** [finagle/finch#136]
The idea is just to provide an implicit view from `OAuth2Request` to `HttpRequest` and put it into `oauth2` module. I'm also open for discussion on how can we make a better integration.

**Comment by ilya-murzinov** [comment]
Hi, @vkostyukov!
I was trying to work on this issue and got stuck in the very beginning.
I added new module to the project, but can't resolve **finch-oauth2** dependency:
```scala
lazy val oauth2 = project
  .settings(moduleName := "finch-oauth2")
  .settings(libraryDependencies += "com.twitter" %% "finagle-oauth2" % "0.1.4")
  .settings(resolvers += "Finagle-OAuth2" at "http://repo.konfettin.ru")
  .settings(allSettings)
  .dependsOn(core)
```

Am I doing something wrong? Why doesn't it work?
Thanks.

**Comment by vkostyukov** [comment]
@ilya-murzinov thanks for looking into this!

Yeah. Some time ago I migrated the `finagle-oauth2` to `finagle-httpx`, but I haven't published the binaries yet. Perhaps I should publish it into the MC under the same (or difference organization).

@travisbrown do we have `io.github.twitter` already. If so we should probably publish it there.

**Comment by travisbrown** [comment]
No, I haven't published anything there yet, but yes, that's probably where it should go. I'm planning to spend the week after next catching up on some of this stuff, if it can wait until then.

**Comment by vkostyukov** [comment]
Good news: I just published the finagle-oauth2 library to MC: should be synced in a couple of hours. Once it's synced we can start working on this issue.

**Comment by vkostyukov** [comment]
I've started working on this.

**Comment by vkostyukov** [comment]
I'm not quite sure yet what to do here. So I'm moving it to the next milestone.

**Comment by vkostyukov** [comment]
Fixed in #419.
